### PR TITLE
Fix launch.json squiggle for "console".

### LINKED ISSUE #7048: "console" property in launch.json for cppdbg squiggles with "Property console is not allowed."
...when generating a cppbuild task.

### PR DESCRIPTION
**Pull request by sean-mcmanus**
Fixes https://github.com/microsoft/vscode-cpptools/issues/7048

### CODE REVIEW

**Review by WardenGnaw — CHANGES_REQUESTED** [review]
`cppdbg` does not use console.

**Comment by sean-mcmanus** [comment]
> `cppdbg` does not use console.

Ah, okay, then there's a different bug.

File: Extension/tools/OptionsSchema.json [hunk 1]
```diff
@@ -561,7 +578,7 @@
             "%c_cpp.debuggers.cppvsdbg.console.newExternalWindow.description%"
           ],
           "description": "%c_cpp.debuggers.cppvsdbg.console.description%",
-          "default": "internalConsole"
+          "default": "externalTerminal"
```
Comment (WardenGnaw) [inline Extension/tools/OptionsSchema.json:581]: This is going to mismatch when the user does not provide `console`. vsdbg will default to `internalConsole`.

Comment (sean-mcmanus) [reply Extension/tools/OptionsSchema.json:581]: @WardenGnaw Why is package.json "default" set to "externalTerminal" then? Isn't that a mismatch?

Comment (sean-mcmanus) [reply Extension/tools/OptionsSchema.json:581]: When I run generateOptionsSchema this gets updated.

**Comment by WardenGnaw** [comment]
The error is:
https://github.com/microsoft/vscode-cpptools/blob/2199ec324d0074253522695863314d52621d83ec/Extension/src/Debugger/configurations.ts#L47

it should be
```
type === "cppdbg ? "externalConsole" : false : "console" : "externalTerminal"
```

**Comment by sean-mcmanus** [comment]
> The error is:
> https://github.com/microsoft/vscode-cpptools/blob/2199ec324d0074253522695863314d52621d83ec/Extension/src/Debugger/configurations.ts#L47
> 
> it should be
> 
> ```
> type === "cppdbg ? "externalConsole" : false : "console" : "externalTerminal"
> ```

Yeah, but something is adding `"console": "externalTerminal"` to the end of the configuration...not sure where that's coming from.

**Comment by WardenGnaw** [comment]
The issue is that `createLaunchString` is used for both `cppvsdbg` and `cppdbg`. It will need to switch based on which configuration it is generating for. 

The snippet above should fix that issue.

**Comment by sean-mcmanus** [comment]
> The issue is that `createLaunchString` is used for both `cppvsdbg` and `cppdbg`. It will need to switch based on which configuration it is generating for.
> 
> The snippet above should fix that issue.

There's another spot in the code -- I found it. I'll submit a 2nd PR soon.

**Pull request closed without merging**